crates/fuel-streams-core/src/nats/nats_client.rs

```diff
@@ -6,60 +6,44 @@ use async_nats::{
         stream,
     },
 };
-use bytes::Bytes;
 use tracing::info;
 
-use super::{types::*, ClientOpts, NatsError, NatsNamespace};
+use super::{types::*, NatsClientOpts, NatsError, NatsNamespace};
 
 #[derive(Debug, Clone)]
+/// NatsClient is a wrapper around the NATS client that provides additional functionality
+/// geared towards fuel-streaming use-cases
 pub struct NatsClient {
     pub nats_client: async_nats::Client,
     pub jetstream: JetStreamContext,
     pub namespace: NatsNamespace,
-    pub opts: ClientOpts,
+    pub opts: NatsClientOpts,
 }
 
 impl NatsClient {
-    pub async fn connect(opts: ClientOpts) -> Result<Self, NatsError> {
-        let url = opts.to_owned().url;
-        let nats_client = opts
-            .connect_opts()
-            .connect(&url)
-            .await
-            .map_err(|e| NatsError::ConnectionError { url, source: e })?;
+    pub async fn connect(opts: &NatsClientOpts) -> Result<Self, NatsError> {
+        let url = &opts.url;
+        let namespace = opts.namespace.clone();
+        let nats_client =
+            opts.connect_opts().connect(&url).await.map_err(|e| {
+                NatsError::ConnectionError {
+                    url: url.to_string(),
+                    source: e,
+                }
+            })?;
 
         let jetstream = async_nats::jetstream::new(nats_client.to_owned());
-        info!("Connected to NATS server at {}", &opts.url);
+        info!("Connected to NATS server at {}", url);
 
         Ok(Self {
             nats_client,
             jetstream,
             opts: opts.to_owned(),
-            namespace: opts.namespace,
+            namespace,
         })
     }
 
-    #[allow(dead_code)]
-    fn validate_payload(
```

Comment:
Why are we dropping this ?

Reply:
It's not used anywhere atm and it would be a concern of `fuel-streams-publisher`, so it should not exist in the `core`

Reply:
Right, does it make sense to move it to the publisher then ? Just asking as the check itself makes total sense and we might need it at some point ?

Reply:
Not at the moment, it's currently more ergonomic to delegate that logic to the NATS infrastructure. We control both the publishing and the Stream infrastructure, so there's no reason we need to validate at runtime before publishing since that's a strong indication of a bad configuration from our side (One of the anomalies I mentioned earlier that we can avoid)